src/main/java/com/exactpro/th2/act/ActHandler.java

```diff
@@ -90,34 +91,34 @@
             MessageRouter<EventBatch> eventBatchRouter,
             Check1Service verifierService
     ) {
-        this.messageRouter = Objects.requireNonNull(router, "'Router' parameter");
-        this.eventBatchMessageRouter = Objects.requireNonNull(eventBatchRouter, "'Event batch router' parameter");
-        this.verifierConnector = Objects.requireNonNull(verifierService, "'Verifier service' parameter");
-        this.callbackList = Objects.requireNonNull(callbackList, "'Callback list' parameter");
+        this.messageRouter = requireNonNull(router, "'Router' parameter");
+        this.eventBatchMessageRouter = requireNonNull(eventBatchRouter, "'Event batch router' parameter");
+        this.verifierConnector = requireNonNull(verifierService, "'Verifier service' parameter");
+        this.callbackList = requireNonNull(callbackList, "'Callback list' parameter");
     }
 
     @Override
     public void placeOrderFIX(PlaceMessageRequest request, StreamObserver<PlaceMessageResponse> responseObserver) {
         try {
-            if (logger.isDebugEnabled()) {
-                logger.debug("placeOrderFIX request: " + shortDebugString(request));
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("placeOrderFIX request: " + shortDebugString(request));
             }
-            placeMessage(request, responseObserver, "NewOrderSingle", "ClOrdID", request.getMessage().getFieldsMap().get("ClOrdID").getSimpleValue(),
-                    ImmutableSet.of("ExecutionReport", "BusinessMessageReject"), "placeOrderFIX");
+            placeMessage(request, responseObserver, "NewOrderSingle", request.getMessage().getFieldsMap().get("ClOrdID").getSimpleValue(),
+                    ImmutableMap.of("ExecutionReport", new CheckMetadata("ClOrdID"), "BusinessMessageReject", new CheckMetadata("BusinessRejectRefID", FAILED)), "placeOrderFIX");
```

Comment:
The only thing I would do is creating factory methods instead of calling the constructor.

Something like that:
```suggestion
                    ImmutableMap.of("ExecutionReport", CheckMetadata.passOn("ClOrdID"), "BusinessMessageReject", CheckMetadata.failOn("BusinessRejectRefID")), "placeOrderFIX");
```